### PR TITLE
[93] User confirmation after signup

### PR DESCRIPTION
**Pull request by katelovescode**
Adds devise views and custom mailer (left in the ones we're not using now because I'm sure we'll need them later), overrides devise confirmation controller to return JSON (and specifically the token), helper rake task for testing/linting/rswag generation before PR, non-rendering React component to make the API call to the confirmation backend, error handling and translation on the front-end

# 💅🏼 What issue does this fix?
#93 

## 🍂 Before You Submit

* [X] Did you write tests?
* [X] Did you run `rails rswag` from the root?
* [X] Did you run `rubocop` from the root?
* [X] Did you run `yarn lint` in `/client`?
* [X] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment Considerations

* [ ] Data Migrations
* [ ] Schema Migrations
* [ ] Dependencies

## 🧵 Steps to set up locally

restart your rails server; might need to restart the frontend too but I don't think so.

## 🧳 Steps to test
- sign up as a new user
- click the confirmation link that opens in the browser
- you should be logged in and redirected to getting-started
- log out
- click the confirmation link again
- you should get an error telling you your user has already been confirmed, and you should try logging in
- log in as the new user
- you should be logged in and redirected to getting-started
- visit `localhost:3000/confirm?confirmation_token=cactus`
- you should get an error indicating your token is invalid and that you should request a new confirmation token (this behavior does not exist)

## 🕯 Caveats, concerns
We've got error handling for expired links but no tests for it, I'd like to write a ticket for it instead of holding this PR up any longer.

Ditto for requesting a new confirmation token.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 Rails.application.routes.draw do
   # This is required because the `devise_for` call generates a `GET /login`
-  # route which we don't want to expose.
+  # route which we don't want to expose; same for confirmation
```
Comment (arku) [inline config/routes.rb:7]: The comment `same for confirmation` doesn't apply anymore.

File: config/environments/test.rb [hunk 2]
```diff
@@ -41,6 +41,7 @@
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3001 }
```
Comment (arku) [inline config/environments/test.rb:44]: A minor thing. Why is this 3001?

File: config/locales/es.yml [hunk 3]
```diff
@@ -2,3 +2,4 @@ es:
   errors:
     messages:
       taken: "ya ha sido tomado"
+      confirmation: "Error de confirmación"
```
Comment (katelovescode) [inline config/locales/es.yml:5]: cc @csprayregen for translation review

File: client/src/i18n/locales/es/index.json [hunk 4]
```diff
@@ -76,6 +76,19 @@
 
   "genericErrorMessage": "Algo salió mal, inténtalo de nuevo más tarde.",
 
+  "alreadyConfirmed": "Este correo electrónico ya ha sido confirmado, por favor intente iniciar sesión. Si ha olvidado tu contraseña, por favor restáurela abajo",
+  "confirmationPeriodExpired": "Tu período de confirmación ha expirado",
```
Comment (katelovescode) [inline client/src/i18n/locales/es/index.json:80]: cc @csprayregen for translation review

File: config/environments/test.rb (hunk 2 above)
Comment (katelovescode) [reply config/environments/test.rb]: Because I copied it directly from the environment.rb when I removed it and made environment specific ones, wheeeee 😎 removed.

File: README.md [hunk 5]
```diff
@@ -101,9 +101,23 @@ Visit `localhost:3001/api-docs` to see Swagger UI for API endpoints 📑
...
 - `bundle exec rspec` or `bundle exec guard` to watch
 - When tests pass and you're ready for a PR, please run `rails rswag` to update the API documentation
 
+### Frontend
+
+- `yarn test` (auto-watch) or `yarn test-once` to run the suite one time only
+
+### End to End
+
+- `yarn cy:ci` from the root directory
```
Comment (arku) [inline README.md:115]: @katelovescode Can we also add a step to build the frontend and copy it to the `public` directory before running `yarn cy:ci`?

File: app/controllers/confirmations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+class ConfirmationsController < Devise::ConfirmationsController
+  respond_to :json
+
+  def new
+    super
+  end
+
+  def create
+    super
+  end
```
Comment (arku) [inline app/controllers/confirmations_controller.rb:13]: **suggestion (non-blocking)**: We don't need to define the `new` and `create` actions as we're not customizing them.

File: app/controllers/confirmations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  def current_token
+    request.env['warden-jwt_auth.token']
+  end
+
+  def error_attribute
+    @error_attribute ||= @errors.keys.first
+  end
+
+  def error_type
+    @error_type ||= @errors[@errors.keys.first].first[:error]
```
Comment (arku) [inline app/controllers/confirmations_controller.rb:54]:
```suggestion
    @error_type ||= @errors[@error_attribute].first[:error]
```

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<%= image_tag attachments['pieFullTanLogo.svg'].url, width: '150', align: "middle" %>
+
+<p>Hi<%= @greeting_name ? " #{@greeting_name}," : ","%></p>
+
+<p>Welcome to Pie for Providers! Get ready to increase your slice of the pie with simple child care subsidy management tools. To get started, please verify your email address by clicking the link below (you can also copy and paste it into your browser).</p>
+
+<p><%= link_to 'Confirm Account', "#{ActionMailer::Base.default_url_options[:protocol]}#{ActionMailer::Base.default_url_options[:host]}#{ActionMailer::Base.default_url_options[:port]}/confirm?confirmation_token=#{@resource.confirmation_token}", target: "_blank" %></p>
```
Comment (arku) [inline app/views/devise/mailer/confirmation_instructions.html.erb:7]: **suggestion (non-blocking)**: Can we move `ActionMailer::Base.default_url_options` to the mailer and share it via `@options` or something?

Comment (arku) [reply app/views/devise/mailer/confirmation_instructions.html.erb:7]: Actually, we can move the whole URL generation to a `private` method and share the confirmation URL via an instance variable.

File: app/views/devise/mailer/email_changed.html.erb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
```
Comment (arku) [inline app/views/devise/mailer/email_changed.html.erb:1]: This is out of scope for this PR. We might want to use the greeting name like done [here](https://github.com/pieforproviders/pieforproviders/pull/208/files#diff-6909b6f56fb271baaecf5e7ef68425d9R3).

File: app/views/devise/mailer/password_change.html.erb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<p>Hello <%= @resource.email %>!</p>
```
Comment (arku) [inline app/views/devise/mailer/password_change.html.erb:1]: Same as https://github.com/pieforproviders/pieforproviders/pull/208/files#r481546872.

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+
+<p>Questions? Email <%= mail_to 'tech@pieforproviders.com', 'tech@pieforproviders.com', subject: 'Pie for Providers: question after signup' %></p>
```
Comment (arku) [inline app/views/devise/mailer/confirmation_instructions.html.erb:9]: 👌

File: app/views/devise/mailer/email_changed.html.erb (hunk 9 above)
Comment (katelovescode) [reply app/views/devise/mailer/email_changed.html.erb:1]: Ya these emails haven't been designed, not sure they're even on the design radar, we should ping @csprayregen about generating tickets for these emails.

File: app/views/devise/mailer/password_change.html.erb (hunk 10 above)
Comment (katelovescode) [reply app/views/devise/mailer/password_change.html.erb:1]: @csprayregen

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<%= image_tag attachments['pieFullTanLogo.svg'].url, width: '150', align: "middle" %>
+
+<p>Hi<%= @greeting_name ? " #{@greeting_name}," : ","%></p>
+
+<p>Welcome to Pie for Providers! Get ready to increase your slice of the pie with simple child care subsidy management tools. To get started, please verify your email address by clicking the link below (you can also copy and paste it into your browser).</p>
```
Comment (katelovescode) [inline app/views/devise/mailer/confirmation_instructions.html.erb:5]: @csprayregen this whole file needs translation

File: app/views/devise/mailer/email_changed.html.erb (hunk 9 above)
Comment (katelovescode) [reply app/views/devise/mailer/email_changed.html.erb:1]: PS. Chelsea, all the mailers that AREN'T confirmation_instructions don't need translation; we don't have a design/layout for them yet I don't think; that's what the tickets should be.

File: app/mailers/devise_custom_mailer.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# Mailer Overrides for Devise default emails
+class DeviseCustomMailer < Devise::Mailer
+  include Devise::Controllers::UrlHelpers # Optional. eg. `confirmation_url`
+  default template_path: 'devise/mailer' # to make sure that your mailer uses the devise views
+
+  def confirmation_instructions(record, token, opts = {})
+    @greeting_name = record.greeting_name || ''
```
Comment (arku) [inline app/mailers/devise_custom_mailer.rb:9]: Can we remove `|| ''` as it makes `@greeting_name` truthy and displays "Hi ," in the email?

image

File: client/src/Login/Login.js [hunk 14]
```diff
@@ -1,16 +1,31 @@
...
+  const location = useLocation()
+
   const [apiError, setApiError] = useState(null)
+
   const { makeRequest } = useApiResponse()
   let history = useHistory()
   const { t, i18n } = useTranslation()
 
+  useEffect(() => {
+    if (location?.state?.error?.status) {
```
Comment (arku) [inline client/src/Login/Login.js:18]: A minor thing: `location` can't be falsy. So, we can remove the first `?`.

File: client/src/Login/Login.js [hunk 15]
```diff
@@ -35,6 +50,48 @@ export function Login() {
...
+        )} ${t('requestNewConfirmation')} `}
+        <a href="www.google.com" target="_blank">
+          Placeholder Link
+        </a>
+      </div>
+    )
+  }
+
+  const errorMessage = ({ attribute, type }) => {
+    switch (true) {
```
Comment (arku) [inline client/src/Login/Login.js:73]: We might want to use the `attribute` here.

```
switch(attribute) {
  case 'email': {
   // ...
  }
}
```

Or we could refactor it a bit. If you prefer to do it in a different PR, no worries.

```js
const errorMessages = {
  email: {
    already_confirmed: () => t('alreadyConfirmed'),
    confirmation_period_expired: () => t('confirmationPeriodExpired'),
    default: () => contactUs({ message: t('genericEmailConfirmationError') })
  },
  confirmation_token: {
    blank: () => <ResendToken type="blank" />,
    invalid: () => <ResendToken type="invalid" />,
    default: () => contactUs({ message: t('genericEmailConfirmationError') })
  },
  default: () => t('genericConfirmationError'),
}

const attributeError = errorMessages[attribute]
if (!attributeError) {
  return errorMessages.default()
}

const fn = attributeError[type] || attributeError.default
return fn()
```

File: client/src/Login/Login.js [hunk 16]
```diff
@@ -44,12 +101,20 @@ export function Login() {
         {t('or')} <span className="uppercase font-bold">{t('login')}</span>
       </p>
 
-      {apiError && (
+      {apiError?.status && (
         <Alert
           className="mb-2"
-          message={apiError.message}
+          message={apiError?.message}
```
Comment (arku) [inline client/src/Login/Login.js:107]: If `apiError?.status` is truthy, `apiError` will have a `message` property. So, we don't need the `?`.

File: swagger/v1/swagger.json [hunk 17]
```diff
@@ -942,6 +942,32 @@
         }
       }
     },
+    "/confirmation": {
+      "get": {
+        "summary": "Signs up a new user; creates the user.",
```
Comment (arku) [inline swagger/v1/swagger.json:947]: Looks like this is taken from the registration spec. We might want to update the summary.

File: package.json [hunk 18]
```diff
@@ -8,21 +8,22 @@
...
-    "format": "prettier --write \"**/*.+(js|jsx|json|yml|yaml|css|md|vue)\"",
-    "install": "yarn --cwd client install",
+    "start": "yarn --cwd client start",
+    "test": "yarn --cwd client test",
+    "test-once": "yarn --cwd client test --watchAll=false",
     "ad": "yarn --cwd client add",
-    "rm": "yarn --cwd client remove"
+    "rm": "yarn --cwd client remove",
+    "install-all": "yarn install && yarn --cwd client install",
+    "format": "prettier --write \"**/*.+(js|jsx|json|yml|yaml|css|md|vue)\""
```
Comment (arku) [inline package.json:26]: `prettier` isn't available in the root `node_modules`. We might want to install it.

File: spec/cypress/integration/confirmation.spec.js [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+        url: `/confirmation?confirmation_token=${confirmationToken}`,
+      }).as('confirmation')
+
+      cy.visit(`/confirm?confirmation_token=${confirmationToken}`)
+      cy.wait('@confirmation')
+      cy.get(createSelector('authError')).should('exist')
+      cy.location('pathname').should('eq', '/login')
+    })
+  })
+})
```
Comment (arku) [inline spec/cypress/integration/confirmation.spec.js:111]: 😍

File: spec/cypress/integration/login.spec.js [hunk 20]
```diff
@@ -20,9 +20,9 @@ describe('Login', () => {
             full_name: fullName,
```
Comment (arku) [inline spec/cypress/integration/login.spec.js:20]: A minor thing: Can we remove the `spec/cypress/videos/login.spec.js.mp4` video?

File: spec/requests/confirmations_spec.rb [hunk 21]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'swagger_helper'
+
+RSpec.describe 'GET /confirmation', type: :request do
+  let(:unconfirmed_user) { create(:user, confirmed_at: nil) }
+  let(:confirmed_user) { create(:confirmed_user) }
+  path '/confirmation' do
+    get 'Signs up a new user; creates the user.' do
```
Comment (arku) [inline spec/requests/confirmations_spec.rb:10]: Looks like this is taken from the registration spec. Can we update it?

File: app/controllers/confirmations_controller.rb [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+class ConfirmationsController < Devise::ConfirmationsController
+  respond_to :json
+
+  def new
+    super
+  end
+
+  def create
+    super
+    @front_end_path = concatenate_path
```
Comment (arku) [inline app/controllers/confirmations_controller.rb:13]: This should be added to [`DeviseCustomMailer#confirmation_instructions`](https://github.com/pieforproviders/pieforproviders/pull/208/files#diff-62839391a18852f6d923c89329ad71ffR8).

File: spec/cypress/integration/login.spec.js (hunk 20 above)
Comment (katelovescode) [reply spec/cypress/integration/login.spec.js:20]: I don't have it locally but I'll remove it outside of this branch.

File: README.md [hunk 23]
```diff
@@ -101,9 +101,24 @@ Visit `localhost:3001/api-docs` to see Swagger UI for API endpoints 📑
...
+- `yarn test` (auto-watch) or `yarn test-once` to run the suite one time only
+
+### End to End
+
+- `yarn build && yarn deploy`
+- `yarn cy:ci` from the root directory
+
+### Prep for Pull Request
+
+There's a helper rake task that runs all test suites and linting steps, and generates the swagger documentation; use `rails prep` to run this command.
```
Comment (weedySeaDragon) [inline README.md:120]: 💯

File: app/controllers/confirmations_controller.rb [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+    "#{options[:protocol]}#{options[:host]}#{options[:port]}"
+  end
+
+  def errors(details = nil)
+    @errors ||= details
+  end
+
+  def error_response
+    {
+      error: I18n.t('errors.messages.confirmation'),
```
Comment (weedySeaDragon) [inline app/controllers/confirmations_controller.rb:39]: This is not using the right locale.
1. sign up in English.
2. (before using a confirmation URL with a bad token)  Go to the login page and switch the language to Spanish.
3. Use a URL with a bad confirmation token
4. The alert displayed does not translate the error message:

confirmation-error-wrong-locale-i18n


The backend (Rails) does not have any info about the locale so uses the default one (`:en`).
The frontend (React) just displays the message without doing any translation.
In this case, the front end _has_ to be responsible for doing the translation since Rails does not have any response header with the locale info prior to the GET (`show` with the `ConfirmationsController`)

(This will be true for any page displayed as the result of a link, yes?)

(I think this is a separate issue/PR but YMMV)

File: app/mailers/devise_custom_mailer.rb [hunk 25]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# Mailer Overrides for Devise default emails
+class DeviseCustomMailer < Devise::Mailer
+  include Devise::Controllers::UrlHelpers # Optional. eg. `confirmation_url`
```
Comment (weedySeaDragon) [inline app/mailers/devise_custom_mailer.rb:5]: why is this needed?

File: app/mailers/devise_custom_mailer.rb [hunk 26]
```diff
@@ -0,0 +1,15 @@
...
+  default template_path: 'devise/mailer' # to make sure that your mailer uses the devise views
+
+  def confirmation_instructions(record, token, opts = {})
+    @greeting_name = record.greeting_name
+    @token = token
+    opts[:subject] = 'Pie for Providers email verification'
+    attachments.inline['pieFullTanLogo.svg'] = File.read(Rails.root.join('app/views/devise/mailer/assets/pieFullTanLogo.svg'))
```
Comment (weedySeaDragon) [inline app/mailers/devise_custom_mailer.rb:12]: why this path?  why isn't the `.svg` under the conventional `app/assets/images...` path?

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 27]
```diff
@@ -0,0 +1,9 @@
...
+
+<p><%= link_to 'Confirm Account', "#{@front_end_path}/confirm?confirmation_token=#{@resource.confirmation_token}", target: "_blank" %></p>
```
Comment (weedySeaDragon) [inline app/views/devise/mailer/confirmation_instructions.html.erb:7]: Should use a helper to construct the url or have it constructed by the mailer and saved as an `@` variable to use in the view

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 28]
```diff
@@ -0,0 +1,9 @@
...
+
+<p>Questions? Email <%= mail_to 'tech@pieforproviders.com', 'tech@pieforproviders.com', subject: 'Pie for Providers: question after signup' %></p>
```
Comment (weedySeaDragon) [inline app/views/devise/mailer/confirmation_instructions.html.erb:9]: These email addresses shouldn't be hardcoded.  (Perhaps get them from ENV or an application configuration value.  An application configuration would not require restarting the server to if the value changes; ENV vars do.)

File: app/controllers/confirmations_controller.rb (hunk 24 above)
Comment (katelovescode) [reply app/controllers/confirmations_controller.rb:33]: @weedySeaDragon Can you write this ticket up, really great catch.

File: app/mailers/devise_custom_mailer.rb (hunk 25 above)
Comment (katelovescode) [reply app/mailers/devise_custom_mailer.rb:5]: It probably isn't.  Was the default from a Custom Mailer template.  I'm gonna leave it until we've finished all the other mailers though, just in case we end up needing them.

File: app/mailers/devise_custom_mailer.rb (hunk 26 above)
Comment (katelovescode) [reply app/mailers/devise_custom_mailer.rb:13]: Because API-only Rails doesn't load assets through the asset pipeline, so it doesn't really matter where it's stored, and I thought it made more sense to put it near the mailers where it's used rather than an asset pipeline folder that isn't being used as it would be in a non-API implementation

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 12 above)
Comment (katelovescode) [reply app/views/devise/mailer/confirmation_instructions.html.erb]: @csprayregen I actually took a swipe at it, please review in the changed code files.

File: app/controllers/confirmations_controller.rb (hunk 24 above)
Comment (katelovescode) [reply app/controllers/confirmations_controller.rb:33]: HOLY CRAP - never mind!  I just found Github's "reference in a new issue" feature and I made the ticket!

File: client/src/i18n/locales/es/index.json [hunk 29]
```diff
@@ -76,6 +76,19 @@
 
   "genericErrorMessage": "Algo salió mal, inténtalo de nuevo más tarde.",
 
+  "alreadyConfirmed": "Este correo electrónico ya ha sido confirmado, por favor intente iniciar sesión. Si ha olvidado tu contraseña, por favor restáurela abajo",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:79]: "alreadyConfirmed": "Este correo electrónico ya ha sido confirmado, por favor intenta iniciar sesión. Si has olvidado tu contraseña, por favor resetéala abajo",

File: client/src/i18n/locales/es/index.json (hunk 4 above)
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:80]: Tu periodo de confirmación ha expirado

File: client/src/i18n/locales/es/index.json [hunk 30]
```diff
@@ -76,6 +76,19 @@
...
+  "confirmationPeriodExpired": "Tu período de confirmación ha expirado",
+  "genericEmailConfirmationError": "Hemos tenido un problema para confirmar tu correo electrónico",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:81]: "genericEmailConfirmationError": "Hemos tenido un problema en confirmar tu correo",

File: client/src/i18n/locales/es/index.json [hunk 31]
```diff
@@ -76,6 +76,19 @@
...
+  "yourConfirmationToken": "Tu símbolo de confirmación es",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:82]: "yourConfirmationToken": "Tu token de confirmación es",

File: client/src/i18n/locales/es/index.json [hunk 32]
```diff
@@ -76,6 +76,19 @@
...
+  "blank": "en blanco",
+  "invalid": "inválido",
+  "mustUseConfirmationEmail": "Debe confirmar tu cuenta utilizando un enlace de confirmación enviado a tu dirección de correo electrónico",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:85]: "mustUseConfirmationEmail": "Tienes que confirmar tu cuenta utilizando un enlace de confirmación enviado a tu correo",

File: client/src/i18n/locales/es/index.json [hunk 33]
```diff
@@ -76,6 +76,19 @@
...
+  "requestNewConfirmation": "Puede solicitar un nuevo correo electrónico de confirmación utilizando este enlace:",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:86]: "requestNewConfirmation": "Puedes solicitar un nuevo correo de confirmación utilizando este enlace:",

File: client/src/i18n/locales/es/index.json [hunk 34]
```diff
@@ -76,6 +76,19 @@
...
+  "genericConfirmationTokenError": "Hemos tenido un problema con tu token de confirmación",
+  "genericConfirmationError": "Hubo un problema desconocido con tu confirmación",
+  "contactUs": "Contáctenos",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:89]: Contáctanos

File: client/src/i18n/locales/es/index.json [hunk 35]
```diff
@@ -76,6 +76,19 @@
...
+  "forSupport": "para el apoyo",
```
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:90]: para el soporte técnico

**Review by csprayregen — COMMENTED** [review]
@katelovescode this is done!

File: client/src/i18n/locales/es/index.json (hunk 32 above)
Comment (csprayregen) [reply client/src/i18n/locales/es/index.json:85]: @katelovescode flagging that this needs to have a period at the end of it. Seeing the earlier version of this translation come up w/o one in my QA of #93

File: config/locales/es.yml [hunk 36]
```diff
@@ -2,3 +2,11 @@ es:
...
+  mailers:
+    confirmation_instructions:
+      hello: Hola
+      body: ¡Bienvenido a Pie for Providers! Prepárate para aumentar tu porción del pastel con simples herramientas de administración de subsidios de cuidado infantil. Para empezar, por favor verifique su dirección de correo electrónico haciendo clic en el siguiente enlace (también puede copiar y pegar en su navegador).
```
Comment (csprayregen) [inline config/locales/es.yml:9]: ¡Bienvenidas a Pie for Providers! Prepárate para aumentar tu porción del pastel con herramientas sencillas de administración del subsidio. Para empezar, por favor verifica tu dirección de correo haciendo clic en el enlace abajo (también puedes copiar y pegar en su navegador).

File: config/locales/es.yml [hunk 37]
```diff
@@ -2,3 +2,11 @@ es:
...
+      confirm_account: Confirmar la cuenta
```
Comment (csprayregen) [inline config/locales/es.yml:10]: Confirma la cuenta

File: config/locales/es.yml [hunk 38]
```diff
@@ -2,3 +2,11 @@ es:
...
+      questions: "¿Preguntas? Correo electrónico "
+      subject: Verificación del correo electrónico de Pie for Providers
```
Comment (csprayregen) [inline config/locales/es.yml:12]: Verificación del correo de Pie for Providers

File: config/locales/es.yml (hunk 3 above)
Comment (csprayregen) [reply config/locales/es.yml:5]: @katelovescode this is done!
